app/Shop/Categories/Repositories/CategoryRepository.php

```diff
@@ -115,12 +115,23 @@ public function updateCategory(array $params) : Category
         }
 
         $merge = $collection->merge(compact('slug', 'cover'));
-        if (isset($params['parent'])) {
+
+        // set parent attribute default value if not set
+        $params['parent'] = $params['parent'] ?? 0;
+
+        // If parent category is not set on update
+        // just make current category as root
+        // else we need to find the parent
+        // and associate it as child
+        if ( (int)$params['parent'] == 0) {
+            $category->saveAsRoot();
+        } else {
             $parent = $this->findCategoryById($params['parent']);
             $category->parent()->associate($parent);
         }
 
         $category->update($merge->all());
+        
```

Comment:
Is the category related fix needed on this PR?

Reply:
No, I did this on a separate branch. Only stubs and command created are needed

Reply:
Yes, We should remove them. Can you do it?